capstone-app/src/app/modules/history/wiki-results.service.ts

```diff
@@ -17,15 +19,74 @@ export class WikiResultsService {
     'https://en.wikipedia.org/w/api.php?origin=*&action=parse&page=';
   private static readonly URL_END = '&format=json';
 
-  search(query: string): Observable<WikiSearchResult> {
+  search(query: string): Observable<{title: string, history: string, furtherReading: Map<string, string>}> {
     return this.searchHandler
       .getQueryString(query)
       .pipe(
         switchMap((result) =>
           this.http.get<WikiSearchResult>(
             `${WikiResultsService.URL_BEGINNING}${result}${WikiResultsService.URL_END}`
           )
-        )
+        ),map(history => {
+          if (history.parse.text) {
+            const historyResult = this.fixString(history.parse.text['*']);
+            const title = history.parse.title;
+            return {title, ...historyResult};
+          }
+          else {
+            throw new Error("API did not return a valid response");
+          }
+        }),
+        catchError(() => {
+          console.log('API did not return a valid response.');
+          const title = '';
+          const history = '';
+          const furtherReading = new Map<string, string>();
+          return of({title, history, furtherReading});
+        })
       );
   }
+
+  fixString(text: string): {history: string, furtherReading: Map<string, string>} {
+    const firstIndex = text.indexOf('<span class="mw-headline" id="History">History</span>', 0);
+    if (firstIndex !== -1){
+      const firstPartOfString = text.substring(firstIndex, text.length);
+      const endIndex = firstPartOfString.indexOf('<h2>', 0);
+      const startIndex = firstPartOfString.indexOf('</h2>', 0);
+      let middleOfString = firstPartOfString.substring(startIndex, endIndex);
+      const paragraphs = middleOfString.split('<p>');
+      if (paragraphs.length > 6){
+        middleOfString = paragraphs.slice(0,5).join('');
+      }
+      const furtherReading = this.findHrefs(middleOfString);
+      let history = middleOfString.split(/<.*?>/g).join('');
+      history = history.split(/&.*?;/g).join('');
+      return {history, furtherReading};
+    }
+    console.log('The city page was found, but unfortunately there was no history paragraph found!');
```

Comment:
Should appear in DOM, not console

Reply:
I don't want to put that in the DOM bc I am still putting the content of the wiki page there, it just might not be the exact history section. Usually this means that the page has very little info at all, but there is still something. I will change it to console.error instead of console.log